### PR TITLE
Network Integration and multiple connections

### PR DESCRIPTION
**Pull request by AntonAsbrink**
This is a very big pull request and i should probably assist in going trough and explaining myself. Additionally how to test all of this.

But for solo testing. Check config that "i_network 1" "s_ip localhost"
Open server in bin/server -> write "1" -> write "AddClient" twice
Start up two instances of sandbox
You should see two cubes in the world updating for each client.

This pull request also contains an update to Options to be more versatile.

### CODE REVIEW

File: StortSpelProjekt/Vendor/config.txt [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+f_volume 1.0
+f_backgroundVolume 1.0
+f_updateRate 30
+i_resolutionWidth 1920
+i_resolutionHeight 1080
+i_windowWidth 1200	
+i_windowHeight 800
+i_fullscreen 0
+i_amountOfBlur 6
+i_network 1
```
Comment (OtakuFreak13) [inline StortSpelProjekt/Vendor/config.txt]: Change i_network to 0 before pushing up to develop.

File: StortSpelProjekt/Sandbox/src/main.cpp [hunk 2]
```diff
@@ -9,6 +10,7 @@ Scene* FredriksTestScene(SceneManager* sm);
 Scene* WilliamsTestScene(SceneManager* sm);
 Scene* AndresTestScene(SceneManager* sm);
 Scene* BjornsTestScene(SceneManager* sm);
+Scene* AntonTestScene(SceneManager* sm);
```
Comment (OtakuFreak13) [inline StortSpelProjekt/Sandbox/src/main.cpp:13]: Change back to someone elses scene before pushing up to develop.

**Review by OtakuFreak13 — CHANGES_REQUESTED** [review]
Changes made according to our discord discussion should be made and then it's just the test stuff that needs changing as per my comments.

File: StortSpelProjekt/Game/src/Gamefiles/GameNetwork.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	GameNetwork();
+
+	void SetScene(Scene* scene);
+	void SetNetwork(Network* network);
+	void SetSceneManager(SceneManager* sceneManager);
+
+private:
+	SceneManager* m_pSceneManager;
+	Scene* m_pScene;
+	Network* m_petwork;
```
Comment (fredrix96) [inline StortSpelProjekt/Game/src/Gamefiles/GameNetwork.h]: Change name

**Review by fredrix96 — CHANGES_REQUESTED** [review]
Change one variable name and make sure that you are using forward declaration

**Review by fredrix96 — APPROVED** [review]

**Review by OtakuFreak13 — APPROVED** [review]
Test runs fine and my requested changes were made

**Pull request merged**